pkg/engine/task/task.go

```diff
@@ -126,6 +130,22 @@ func newPipe(task *v1beta1.Task) (Tasker, error) {
 	}, nil
 }
 
+func newToggle(task *v1beta1.Task) (Tasker, error) {
+	// validate if resources are present
+	if len(task.Spec.ResourceTaskSpec.Resources) == 0 {
```

Comment:
Even if Go rules for calling embed fields allows this, I'd still prefer:
```suggestion
	if len(task.Spec.ToggleTaskSpec.Resources) == 0 {
```